### PR TITLE
BUILD-469: add webhook image to csi-driver-shared-resource ci configuration

### PR DESCRIPTION
**Pull request by jkhelil**
referes to https://issues.redhat.com/browse/BUILD-469
@gabemontero PTAL

### CODE REVIEW

File: ci-operator/config/openshift/csi-driver-shared-resource/openshift-csi-driver-shared-resource-master.yaml [hunk 1]
```diff
@@ -15,16 +15,19 @@ binary_build_commands: make build
 build_root:
   from_repository: true
 images:
-- dockerfile_path: Dockerfile
+- dockerfile_path: images/Dockerfile
```
Comment (gabemontero) [inline ci-operator/config/openshift/csi-driver-shared-resource/openshift-csi-driver-shared-resource-master.yaml:18]: yeah @jkhelil you should copy what we do for jenkins

see https://github.com/openshift/release/blob/667722ec35fc0d6ef97846a12b81b49f76702223/ci-operator/config/openshift/jenkins/openshift-jenkins-master.yaml#L17-L18

you should have something like 
```
- context_dir: "images"
  dockerfile_path: Dockerfile
```

and the same for must gather and webhook images

**Comment by gabemontero** [comment]
bit of a "chicken or egg" situation here .... the rehearsals here will not pass until https://github.com/openshift/csi-driver-shared-resource/pull/92 merges, but that PR will not pass its images job until this one does

so we'll have to merge this even though the rehearsals fail

that said, after you make the changes for my comments @jkhelil you will need to run `make openshift-image-mirror-mappings` and then store those changes in a separate commit; also not, to get approval for those mirror mapping changes we'll have to pull in additional approvers outside our team.

File: ci-operator/config/openshift/csi-driver-shared-resource/openshift-csi-driver-shared-resource-master.yaml (hunk 1 above)
Comment (jkhelil) [reply ci-operator/config/openshift/csi-driver-shared-resource/openshift-csi-driver-shared-resource-master.yaml]: fixed

**Comment by jkhelil** [comment]
> bit of a "chicken or egg" situation here .... the rehearsals here will not pass until [openshift/csi-driver-shared-resource#92](https://github.com/openshift/csi-driver-shared-resource/pull/92) merges, but that PR will not pass its images job until this one does
> 
> so we'll have to merge this even though the rehearsals fail
> 
> that said, after you make the changes for my comments @jkhelil you will need to run `make openshift-image-mirror-mappings` and then store those changes in a separate commit; also not, to get approval for those mirror mapping changes we'll have to pull in additional approvers outside our team.

done

**Comment by gabemontero** [comment]
/retitle BUILD-469: add webhook image to csi-driver-shared-resource ci configuration

**Comment by gabemontero** [comment]
/lgtm

/assign @bparees 

for approve of mirror update for our new webhook image here

note https://github.com/openshift/release/pull/27887#issuecomment-1101368231 on rehearsal failures .... once actual repo changes merge we should be OK

**Pull request merged**